### DISCUSSION TITLE
Reflections on Migrating my SaaS to SvelteKit

### DISCUSSION THREAD

**Discussion by Glench** [sveltejs/kit#9667] [Show and tell]
I just migrated my SaaS to SvelteKit and wrote a little about the process. I thought y'all might like to hear my experiences: https://extensionpay.com/articles/migrate-saas-to-sveltekit

And if there's anything you'd like to know I'm happy to answer questions, too.

**Comment by Krowemoh** [comment]
A very friendly post :) I had the same issues with external javascript but I wrapped it inside a setinterval and kept checking if the script was loaded. I see now the onload function would have been perfect. I also think sveltekit seems to almost require using a GUI editor as trying to maintain the nesting structure in my head is a giant pain in the ass. I need to be one level removed from the file I'm trying to edit because vim and + signs don't work. 

The big thing I would love to see is websockets but overall, I'm also very optimistic about sveltekits future. Though I hope it never has a big change like the routing again!

**Reply by Glench** [reply]
Haha yes to all that. You didn't come across any ways to make editing SvelteKit files in vim better? And relatedly, are you using vim plugins  for TypeScript support? The ones I'm currently using don't use the language server stuff and also don't syntax highlight typescript in svelte components so development is not as good as it could be.

> The big thing I would love to see is websockets but overall, I'm also very optimistic about sveltekits future. Though I hope it never has a big change like the routing again!

Ooh yes, websockets would be awesome! I didn't realize SvelteKit didn't support them within the framework. I would also love to see a good story for internationalization / localization since my app is used worldwide.
